lib/units/device/plugins/connect.js

```diff
@@ -1,213 +1,129 @@
 import util from 'util'
 import syrup from '@devicefarmer/stf-syrup'
-import Promise from 'bluebird'
 import logger from '../../../util/logger.js'
-import * as grouputil from '../../../util/grouputil.js'
 import wire from '../../../wire/index.js'
 import wireutil from '../../../wire/util.js'
 import lifecycle from '../../../util/lifecycle.js'
 import db from '../../../db/index.js'
-import dbapi from '../../../db/api.js'
 import adb from '../support/adb.js'
-import router from '../../base-device/support/router.js'
+import connector, {DEVICE_TYPE} from '../../base-device/support/connector.js'
 import push from '../../base-device/support/push.js'
+import router from '../../base-device/support/push.js'
 import group from './group.js'
 import solo from './solo.js'
-import urlformat from './util/urlformat.js'
+import urlformat from '../../base-device/support/urlformat.js'
+import identity from './util/identity.js'
+import data from './util/data.js'
+
+const promiseTimeout = (promise, ms, message = 'Timeout exceeded') => Promise.race([
```

Comment:
This should be explicitly marked with a comment that the promise is not cancelled. Only that the resolve and reject of that promise won't succeed. 
And the promise should not produce any effects without a .then or similar.


Timeouts are a pain in the ass with modern ecmascript..